src/main/java/com/sourcegraph/vcs/RepoUtil.java

```diff
@@ -223,11 +224,22 @@ public static VCSType getVcsType(@NotNull Project project, @NotNull VirtualFile
   }
 
   private static Optional<VirtualFile> getRootFileFromFirstGitRepository(@NotNull Project project) {
+    // https://intellij-support.jetbrains.com/hc/en-us/community/posts/206105769/comments/206091565
+    Object lock = new Object();
+    ProjectLevelVcsManagerImpl.getInstanceImpl(project)
+        .runAfterInitialization(
+            () -> {
+              synchronized (lock) {
+                lock.notify();
+              }
+            });
+    synchronized (lock) {
+      try {
+        lock.wait();
+      } catch (InterruptedException ignored) {
+      }
+    }
```

Comment:
I did some research to see if there was a better solution and didn't find any other official way similar to this "lock & wait". As I understand it, the correct approach would be to hook into the appropriate event, but this would require rewriting part of this repo identification code to be asynchronous. I think we can accept the proposed solution as a temporary one.

BTW is there difference between `ProjectLevelVcsManager.getInstance(project)` and `ProjectLevelVcsManagerImpl.getInstanceImpl(project)`?

Reply:
I can switch it back to `ProjectLevelVcsManager`, originally was testing with some methods only exposed on the class but not needed anymore